tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -694,28 +694,35 @@ def set_timeout(self):
             self.logger.error(msg)
             return ("INTERNAL-ERROR", msg)
 
+
     def run_func(self):
         try:
             self.result = True, self.func(self.protocol, self.url, self.timeout)
         except (webdriver_error.TimeoutException, webdriver_error.ScriptTimeoutException):
             self.result = False, ("EXTERNAL-TIMEOUT", None)
-        except socket.timeout:
-            # Checking if the browser is alive below is likely to hang, so mark
-            # this case as a CRASH unconditionally.
-            self.result = False, ("CRASH", None)
         except Exception as e:
-            if (isinstance(e, webdriver_error.WebDriverException) and
-                    e.http_status == 408 and
-                    e.status_code == "asynchronous script timeout"):
-                # workaround for https://bugs.chromium.org/p/chromedriver/issues/detail?id=2001
-                self.result = False, ("EXTERNAL-TIMEOUT", None)
-            else:
+            status, message = None, None
+            if isinstance(e, socket.timeout):
+                # Checking if the browser is alive in this case is likely to hang,
+                # so mark it as a CRASH unconditionally.
+                status = "CRASH"
+            elif isinstance(e, webdriver_error.WebDriverException):
+                if e.http_status == 408 and e.status_code == "asynchronous script timeout":
+                    # workaround for https://bugs.chromium.org/p/chromedriver/issues/detail?id=2001
+                    status = "EXTERNAL-TIMEOUT"
+                elif e.http_status == 500 and e.status_code == "disconnected":
```

Comment:
`disconnected` is [Chrome-specific], so it probably technically shouldn't live in this file. But since overriding `Executor.convert_result()` isn't easy we can keep this for now.

Unrelated follow-up: I think we can get rid of the `asynchronous script timeout` case above, since the bug has been fixed and there are [no hits] in the code.

[Chrome-specific]: https://source.chromium.org/chromium/chromium/src/+/main:chrome/test/chromedriver/chrome/status.h;l=43;drc=6302d6c3dfa383d38f75835b9d6f84fc826f1fc6;bpv=0;bpt=0
[no hits]: https://source.chromium.org/search?q=%22asynchronous%20script%20timeout%22%20f:chromedriver

Reply:
Yeah, this isn't listed in https://w3c.github.io/webdriver/#dfn-error-code as an error code in the spec.

This is related to https://github.com/w3c/webdriver/issues/1308 about what WebDriver should do given things like content process crashes, which seems like someone from the Chrome team should comment on if Chrome is already returning a non-standard error code here.

Having an error code that no other browser knows about here seems to have little risk, but the comment here should probably acknowledge it's a non-standard Chrome addition.

Reply:
Thanks for the info. Updated the comment and left a todo.